### PR TITLE
Add optional {min,max}imum_uid parameters

### PR DESCRIPTION
**Pull request by g0hl1n**
This PR adds support for optional minimum_uid and maximum_uid parameters to limit allowed UIDs.

### CODE REVIEW

File: pam_ldapdb.cpp [hunk 1]
```diff
@@ -113,10 +117,38 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t* pamh,
...
+        }
+    }
+    if (arguments.find("maximum_uid") != arguments.end()) {
+        if (!pwd) {
+            pam_syslog(pamh, LOG_NOTICE, "unable to get uid for user %s", user);
+            return PAM_AUTH_ERR;
+        }
+
+        maximum_uid = (uid_t) std::stoul(arguments["maximum_uid"]);
+        if (maximum_uid < pwd->pw_uid) {
```
Comment (smackysnacks) [inline pam_ldapdb.cpp]: The readme states `pw_uid < max`, but the code here enforces `pw_uid <= max`

File: pam_ldapdb.cpp [hunk 2]
```diff
@@ -113,10 +117,38 @@ PAM_EXTERN int pam_sm_authenticate(pam_handle_t* pamh,
...
 
+    // check minimum & maximum uid if required
+    pwd = getpwnam(user);
+    if (arguments.find("minimum_uid") != arguments.end()) {
+        if (!pwd) {
+            pam_syslog(pamh, LOG_NOTICE, "unable to get uid for user %s", user);
+            return PAM_AUTH_ERR;
+        }
+
+        minimum_uid = (uid_t) std::stoul(arguments["minimum_uid"]);
```
Comment (smackysnacks) [inline pam_ldapdb.cpp]: `std::stoul` can throw `std::invalid_argument` or `std::out_of_range`. What do you think about catching these and logging them with `pam_syslog`?

Comment (g0hl1n) [reply pam_ldapdb.cpp]: You're right. I'll send an updated PR shortly.

**Comment by g0hl1n** [comment]
Updated PR has landed :-)

**Comment by smackysnacks** [comment]
Thanks!

**Pull request merged**